30412/Tise Tudor/Assignment 1/listManagement.c

```diff
@@ -0,0 +1,197 @@
+//
+// Created by Tudor Tise on 02/03/2020.
+//
+
+#include "listManagement.h"
+
+void setUpFile(){
+    g = fopen("output.dat", "w");
+}
+
+void initializeList()
+{
+    first = NULL;
+    last = NULL;
+}
+
+int getNoOfItems(){
+    NodeT * currentElement = first;
+    int i = 0;
+    while(currentElement != NULL)
+    {
+        currentElement = currentElement->next;
+        i++;
+    }
+    return i;
+}
+
+void addLast(int data)
+{
+    if(first == NULL)
+    {
+        first = (NodeT*)malloc(sizeof(NodeT));
+        first->next = last;
+        first->data = data;
+        last = first;
+    }
+    else
+    {
+        NodeT * newElement = (NodeT*)malloc(sizeof(NodeT));
+        last->next = newElement;
+        newElement->next = NULL;
+        newElement->data = data;
+        last = newElement;
+    }
+}
+
+void addFirst(int data)
+{
+    if(first == NULL)
+    {
+        first = (NodeT*)malloc(sizeof(NodeT));
+        first->next = last;
+        first->data = data;
+        last = first;
+    }
+    else {
+        NodeT *newElement = (NodeT *) malloc(sizeof(NodeT));
+        newElement->data = data;
+        newElement->next = first;
+        first = newElement;
+    }
+}
+
+void removeLast(){
+    if(first == NULL)
+    {
+        fprintf(g,"List is empty!\n");
+    }
+    else {
+        NodeT * currentElement = first->next;
+        while (currentElement->next != NULL){
+            first = currentElement;
+            currentElement = currentElement->next;
+        }
+        free(first->next);
+        first->next = NULL;
+    }
+}
+
+void removeFirst(){
+    if(first == NULL)
+    {
+        fprintf(g,"List is empty!\n");
+    }
+    else {
+        NodeT * currentElement = first;
+        first = first->next;
+        free(currentElement);
+    }
+}
+
+void printFirst(int data){
+    if(first == NULL)
+    {
+        fprintf(g,"List is empty!\n");
+    }
+    else if (data > getNoOfItems()){
+        printAll();
+    } else
+    {
+        NodeT * currentElement = first;
+        for (int i = 0; i<data; i++)
+        {
+            fprintf(g,"%d ", currentElement->data);
+            currentElement = currentElement->next;
+        }
+        fprintf(g,"\n");
+    }
+}
+
+void printLast(int data){
```

Comment:
Wouldn't really name this parameter `data`, it can really have much better names, like `numberOfItemsToPrint` or somewhere along those lines

Reply:
Ok, thank you. I'll be more careful next time.